clang/lib/CodeGen/CGDebugInfo.cpp

```diff
@@ -667,11 +667,8 @@ void CGDebugInfo::CreateCompileUnit() {
   if (CSKind)
     CSInfo.emplace(*CSKind, Checksum);
 
-  if (!CGM.getCodeGenOpts().SYCLUseMainFileName)
-    MainFileDir = getCurrentDirname();
-
   llvm::DIFile *CUFile =
-      DBuilder.createFile(remapDIPath(MainFileName), remapDIPath(MainFileDir),
+      DBuilder.createFile(remapDIPath(MainFileName), remapDIPath(getCurrentDirname()),
```

Comment:
I see this was guarded by `SYCLUseMainFileName` earlier. Can you explain what the option is and why that guard was incorrect please?